### PR TITLE
Upgrade guide to release-6

### PR DESCRIPTION
**Pull request by amarts**
Signed-off-by: Amar Tumballi <amarts@redhat.com>

### CODE REVIEW

File: docs/Upgrade-Guide/upgrade_to_6.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+## Upgrade procedure to Gluster 6, from Gluster 5.x, 4.1.x, and 3.12.x
```
Comment (ShyamsundarR) [inline docs/Upgrade-Guide/upgrade_to_6.md]: Don't we need to add this file to the mkdocs.yml file at the root of the repo? I always thought the build depended on the same.

File: docs/Upgrade-Guide/upgrade_to_6.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+## Deprecated translator upgrade guide
+
+With glusterfs-6.0 release, we are planning to deprecate few of the
+features (or volume types) which existed before.
+
+If you have any one of the below volume types, please go through
+instructions below, so you can preserve sanity in your glusterfs
+release.
+
+### Stripe volume
```
Comment (ShyamsundarR) [inline docs/Upgrade-Guide/upgrade_to_6.md]: I think we need some rewording here as and like so in other cases as well:

Stripe xlator, provided the ability to stripe data across bricks. This functionality was used to create and support files larger than a single brick and also to provide better disk utilization across large file IO, by spread the IO blocks across bricks and hence disks.

This functionality is now provided by the shard xlator (any links here wold help, to configure and use shard).

There is no in place upgrade feasible for volumes using the stripe feature, and users are encouraged to migrate their data from existing stripe based volumes to sharded volumes.

File: docs/Upgrade-Guide/upgrade_to_6.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+data on top of the mount point (like any other rsync like tool),
+to the new volume mount point.
+
+Once you copy over the data, you can delete the volume, and now you
+are set for the upgrade to `glusterfs-6.0`.
+
+
+### Tier volume
+
+If you are using Tier feature, this is not supported in glusterfs-6.0.
```
Comment (ShyamsundarR) [inline docs/Upgrade-Guide/upgrade_to_6.md]: Here the salient bullets would be:

- Tier no longer supported
- No replacement feature in the release
- Convert tiered volumes to regular volumes by detaching the tier
- Then upgrade

File: docs/Upgrade-Guide/upgrade_to_6.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+### Misc
+
+* BD xlator
+* glupy
+* experimental/*
+
+All the above translators were not supported in previous versions, but
+users had an option to create the volumes. Now, if you had created these
+volumes, please delete these volumes and then upgrade to glusterfs-6.0
```
Comment (ShyamsundarR) [inline docs/Upgrade-Guide/upgrade_to_6.md]: migrate data out of these volumes, not delete the volumes :)

File: docs/Upgrade-Guide/upgrade_to_6.md (hunk 2 above)
Comment (amarts) [reply docs/Upgrade-Guide/upgrade_to_6.md]: Done

File: docs/Upgrade-Guide/upgrade_to_6.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+> **NOTE:** Upgrade procedure remains the same as with 4.1.x release
+
+Refer, to the [Upgrading to 4.1](./upgrade_to_4.1.md) guide and follow
+documented instructions, replacing 6 when you encounter 4.1 in the guide as the
+version reference.
+
+## Deprecated translator upgrade guide
+
+With glusterfs-6.0 release, we are planning to deprecate few of the
```
[the remark this answers is not in the document]
Comment (amarts) [reply docs/Upgrade-Guide/upgrade_to_6.md]: Done

File: docs/Upgrade-Guide/upgrade_to_6.md (hunk 1 above)
Comment (amarts) [reply docs/Upgrade-Guide/upgrade_to_6.md]: Done

File: docs/Upgrade-Guide/upgrade_to_6.md (hunk 4 above)
Comment (amarts) [reply docs/Upgrade-Guide/upgrade_to_6.md]: Done

File: docs/Upgrade-Guide/upgrade_to_6.md (hunk 3 above)
Comment (amarts) [reply docs/Upgrade-Guide/upgrade_to_6.md]: Done

**Pull request merged**